refactor: Modualize the samples so that they can be gonew

ikawaha opened the pull request:
see. https://github.com/goadesign/examples/issues/131

* Modularized each sample using Go modules.
    *Temporarily replaced the library with `goa.design/examples/basic` the tracing sample. This library replacement is temporary. After this PR is merged and released, we'll need to revert the replacement.

* Update README with gonew cloning instructions. https://github.com/goadesign/examples/pull/130/commits/f700e779f6c40fa7798022234eac3db15183d390

raphael commented on README.md:
````diff
@@ -7,6 +7,14 @@ This repository contains examples of microservices implemented using
+
+```shell
+$ go install golang.org/x/tools/cmd/gonew@latest
+$ gonew github.com/ikawaha/examples/basic@latest github.com/<your_repo>/basic
````
should this be `gonew goa.design/examples/basic@latest github.com/<your_repo>/basic`?

ikawaha replied:
ooops, fixed in https://github.com/goadesign/examples/pull/130/commits/e0823af5012140691602c7e782994fe488c4a6e0

raphael commented on tracing/go.mod:
```diff
@@ -0,0 +1,26 @@
+go 1.21.3
+
+// Temporarily replacing as the official release is not yet available
+replace goa.design/examples/basic => github.com/ikawaha/calc v0.0.3
```
Should this be removed?

ikawaha replied:
Yes, this `replace` directive is temporary until the official release of `goa.design/examples/basic` becomes available. Once the official release is out, we can remove this replacement.

ikawaha replied:
Would it be okay to remove it in this PR, even if it might cause the tests to fail?

ikawaha replied:
To address your feedback regarding the replace directive, I've made adjustments to ensure a smoother build process. Specifically, I've removed the `tracing` sample from this PR (https://github.com/goadesign/examples/pull/130/commits/772a41ac641d5bad0fc12a3b37a49078b0123c8d). Instead, I've prepared a separate PR specifically for adding the tracing sample(https://github.com/ikawaha/examples/pull/1). This eliminates the need to push the noted replace directive. If it works for you, could we merge this PR first and then move on to merging the subsequent PR for `tracing`?

raphael replied:
Perfect, this looks great.

ikawaha commented:
I've updated the README. I've added information about the Goa Clue example and mentioned the use of the gonew command for easy cloning. [d9ffbdc](https://github.com/goadesign/examples/pull/130/commits/d9ffbdce733e62214d9a59b30f284f5b29747b99)

raphael commented:
Thank you for the great contribution!
